### PR TITLE
Fix output minimization

### PR DESCRIPTION
**Pull request by jeremyrsmith**
- Move minimization of output to notebook encoding time, not on each result
- Use a faster mechanism for minimizing output

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/kernel/Result.scala [hunk 1]
```diff
@@ -189,21 +189,44 @@ object Result {
...
+      val (scanned, last) = rest.foldLeft((Queue.empty[Result], first)) {
+        case ((results, focus), next) =>
+          (focus, next) match {
+            case (Output("text/plain; rel=stdout", focusText), Output("text/plain; rel=stdout", nextText)) =>
+              (results, Output("text/plain; rel=stdout", focusText + nextText))
+            case (a, b) =>
+              (results.enqueue(a), b)
+          }
+      }
+      scanned.enqueue(last).view.map {
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/Result.scala:206]: Why a `view` here? I've always been a bit confused about when it's actually useful to use it.

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/Result.scala:206]: it's supposed to let you do certain seq-ish things without allocating intermediate collections. So if I `view.map(...).toList` then I wouldn't get an intermediate `List` which I would if I did `.toList.map(...)`.

Whether it works well or not I'm not sure...

Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/Result.scala:206]: Hmm, what about `scanned.enqueue(last).map().toList`?

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/Result.scala:206]: That would create an intermediate `Queue` 😄

Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/Result.scala:206]: So essentially whenever you have > 1 operation on a collection, use `view`?

**Comment by jeremyrsmith** [comment]
Closing in favor of #898

**Pull request closed without merging**